### PR TITLE
Add a function that will allow downloading the correct version of env-var-docs.json

### PR DESCRIPTION
**Pull request by jessbumblebee**
This change ads a function that can get the commit sha for a specific version.
In the future, after #132 is submitted, this should be used in config_loader to download the correct version of 
en-var-docs.json. The version that is downloaded in #132 hardcodes a commit sha, which will break in the future
when en-var-docs has changed.

A second similar change will be needed to also download the correct version of the python package in the run script, which installs the python dependencies.

- - - - - - - - - - - - - - - - - -

**Updates from Rocky:**
The commit sha for the tag is different than the commit sha for the commit at the tip of the release, so we need an additional request to the github api to get the correct versino of `env-var-docs.json`. I added this and refactored the parent method a bit to make it more readable (the nested if statements were getting hard to read). 

I tested this locally by deploying 'latest' and deploying with it pinned to a version. I tested v1.24.0 which didn't have `CLIENT_IP_TYPE` in `env-var-docs.json` to make sure that var wouldn't show up even if I had it in my config.

### CODE REVIEW

File: cloud/shared/bin/lib/config_loader.py
Comment (rockycodes) [inline cloud/shared/bin/lib/config_loader.py:153]: question (non-blocking): why assign this to newtag?

File: cloud/shared/bin/lib/config_loader_test.py
Comment (rockycodes) [inline cloud/shared/bin/lib/config_loader_test.py:142]:
```suggestion
        # mock out getting the version number that matches "latest"
```

File: cloud/shared/bin/lib/config_loader_test.py
Comment (rockycodes) [inline cloud/shared/bin/lib/config_loader_test.py:167]: suggestion (blocking): delete or comment back in.

File: cloud/shared/bin/lib/config_loader_test.py
Comment (rockycodes) [inline cloud/shared/bin/lib/config_loader_test.py:179]: suggestion (non-blocking): add a test case for passing in a tag other than "latest" (could use "v1.23.0" from your mock)

**Review by rockycodes — CHANGES_REQUESTED** [review]

**Review by bion — APPROVED** [review]
Overall looks good! Just some cleanup stuff. Approving so you can merge as desired without waiting for another review.

**Review by rockycodes — APPROVED** [review]

File: cloud/shared/bin/env-var-docs-python-dependencies.txt
Comment (rockycodes) [reply cloud/shared/bin/env-var-docs-python-dependencies.txt:2]: @bion I wasn't sure how to do this.

File: cloud/shared/bin/lib/config_loader.py
Comment (rockycodes) [reply cloud/shared/bin/lib/config_loader.py]: done

File: cloud/shared/bin/lib/config_loader.py
Comment (rockycodes) [reply cloud/shared/bin/lib/config_loader.py]: I opted not to extract these url strings to constants since they are not reused.

File: cloud/shared/bin/lib/config_loader.py
Comment (rockycodes) [reply cloud/shared/bin/lib/config_loader.py]: done

File: cloud/shared/bin/env-var-docs-python-dependencies.txt
Comment (rockycodes) [reply cloud/shared/bin/env-var-docs-python-dependencies.txt]: done!

**Review by bion — APPROVED** [review]

**Pull request merged**
